vm_backtrace.c

```diff
@@ -1360,6 +1360,14 @@ rb_profile_frames(int start, int limit, VALUE *buff, int *lines)
 
 	    i++;
 	}
+        else {
+	    cme = rb_vm_frame_method_entry(cfp);
+	    if (cme && cme->def->type == VM_METHOD_TYPE_CFUNC) {
+		buff[i] = (VALUE)cme;
```

Comment:
You forgot to handle the `start` argument.  Although, it was already effectively being ignored (https://bugs.ruby-lang.org/issues/14607).  I've updated https://github.com/ruby/ruby/pull/2713 to also fix the handling of the start argument for this branch.